utils/build.ps1

```diff
@@ -66,6 +66,9 @@ If no such Windows SDK is installed, it will be downloaded from nuget.
 Include the ds2 remote debug server in the SDK.
 This component is currently only supported in Android builds.
 
+.PARAMETER IncludeSanitizers
```

Comment:
Yes agree in principle, but it will also add extra requirements for the default build. For example, the Visual Studio cross libs for Arm64 must be installed (which I had to do explicitly). We need to check if CI is prepared for that and maybe add a note in the quick-start docs right? If we keep it disabled by default, that wouldn't be necessary (now). I'd keep the option for a transition period and see. However, if you want it done now, no problem.

Reply:
Renamed it to `-BuildSanitizers`, so it's easier to distinguish from `-UseSanitizers` option in https://github.com/swiftlang/swift/pull/77844

Reply:
Don't sanitizers add overhead that will impact compiler perf if they are unconditionally used?

Reply:
I think the requirement that the VS cross libs for Arm64 must be installed isn't a big deal as the swift-build instruction already includes that and the swift-build and swift.org CIs already do so, if we need to require that.

If we use `-BuildSanitizers`, the produced compiler doesn't build in sanitizer mode by default, right? If we want to build the toolchain with sanitizers, we'd need to build a second-gen toolchain, that is, build with the sanitizer-built toolchain as the pinned toolchain, right? If so, is there really a compiler performance overhead if we enable `-UseSanitizers` by default? I'd imagine it'd only mean a larger toolchain size mostly due to the additional libraries/DLLs and gives us an option to build in sanitizer mode if we choose to? No?

Reply:
I was able to build and test the small asan example with this. Any reason not to merge this?

Reply:
> the requirement that the VS cross libs for Arm64 must be installed isn't a big deal as the swift-build instruction already includes that

Great. Let me enable `-BuildSanitizers` by default and double-check in PR tests.

> we'd need to build a second-gen toolchain

Exactly

> is there really a compiler performance overhead if we enable -UseSanitizers by default?

Well, binaries grow due to extra sections for the sanitizer counters, which increases link times. And runtime of tests will be longer. I didn't measure yet, but I'd expect a 10-50% slowdown. I think it's good for 2nd-stage test bots like https://ci.swift.org/job/oss-swift-incremental-ASAN-RA-macos/ but not for pre-merge testing.

Reply:
> is there really a compiler performance overhead if we enable -UseSanitizers by default?
Sorry I meant `-BuildSanitizers` rather than `-UseSanitizer`

Reply:
Is there a reason to have `-BuildSanitizers` at all? Can we not just drop the option and enable it unconditionally?